classes/PSElementTrait.php

```diff
@@ -226,6 +226,27 @@ public function sync()
     {
         SRBLogger::addLog('SYNCHRONIZING ' . self::getObjectTypeForMapping() . ' "' . $this->getReference() . '"', SRBLogger::INFO, self::getObjectTypeForMapping(), $this->getDBId());
 
+        // To manage product duplication
+        //TODO generalize to other Elements
+        if (static::getObjectTypeForMapping() === 'product') {
+            $itemsByReference = static::getManyByIdentifier($this->getReference());
+            $countItemsByReference = count($itemsByReference);
+            if ($countItemsByReference > 1) {
+                for ($i = 1; $i < $countItemsByReference; $i++) {
+                    $itemsByReference[$i]->reference = $itemsByReference[$i]->reference . '_' . (microtime(true) * 10000);
```

Comment:
And how does the retailer can now which SRB product is which product in his database ?
Does a shipback with a duplicated product will handle it properly ?

Reply:
The query to get the Products has a sort by ID. So the duplicated product will keep the reference while the duplicate will have a new one. 
The retailer can still recognize its products with their label and all the other attributes. Plus on the SRB's tab, he will see all his Products with their label and their reference, so he will see the changes.

For the Shipback, the changed Products are sync right after getting a new reference, so the SRB database will know which product is which.